axiom/datasets.go

```diff
@@ -281,6 +287,26 @@ func (s *DatasetsService) Info(ctx context.Context, id string) (*DatasetInfo, er
 	return &res, nil
 }
 
+// Trim the dataset identified by its id to a given length. The max duration
+// given will mark the oldest timestamp an event can have. Older ones will be
+// deleted from the dataset.
+func (s *DatasetsService) Trim(ctx context.Context, id string, maxDuration time.Duration) (*TrimResult, error) {
+	req := struct {
```

Comment:
Can we make this a proper struct? Can be internal obv

Reply:
Why? I think this is a nice way of keeping some internals from that function out of the package scope.

Reply:
On the other hand: This will portably extend in the future so you are kinda right.